entities-viewings-collector/src/main/scala/io/renku/entities/viewings/search/DatasetQuery.scala

```diff
@@ -0,0 +1,129 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.entities.viewings.search
+
+import eu.timepit.refined.auto._
+import io.renku.entities.viewings.search.RecentEntitiesFinder.{Criteria, EntityType}
+import io.renku.graph.model.entities.Person
+import io.renku.graph.model.{GraphClass, Schemas}
+import io.renku.triplesstore.SparqlQuery
+import io.renku.triplesstore.SparqlQuery.Prefixes
+import io.renku.triplesstore.client.syntax.FragmentStringContext
+
+object DatasetQuery extends (Criteria => Option[SparqlQuery]) {
+  private[this] val v = Variables.Dataset
+
+  def apply(criteria: Criteria): Option[SparqlQuery] =
+    Option.when(criteria.forType(EntityType.Dataset))(makeQuery(criteria))
+
+  def makeQuery(criteria: Criteria): SparqlQuery =
+    SparqlQuery.of(
+      name = "recent-entity projects",
+      Prefixes.of(Schemas.prov -> "prov", Schemas.renku -> "renku", Schemas.schema -> "schema", Schemas.xsd -> "xsd"),
+      sparql"""|SELECT DISTINCT
+               |  ${v.all}
+               |
+               |  WHERE {
+               |    Bind(1.0 AS ${v.matchingScore})
+               |    Bind('dataset' as ${v.entityType})
+               |
+               |    Graph ${GraphClass.PersonViewings.id} {
+               |      ?personId a renku:PersonViewing;
+               |                renku:viewedDataset ?viewedDataset.
+               |
+               |      ?viewedDataset renku:dataset ?datasetId;
+               |                     renku:dateViewed ${v.viewedDate}
+               |    }
+               |    Graph schema:Dataset {
+               |      ${v.datasetSameAs} a renku:DiscoverableDataset;
+               |                         renku:datasetProjectLink ?projectLink.
+               |
+               |      ?projectLink renku:dataset ?datasetId;
+               |                    renku:project ?projectId.
+               |    }
+               |    
+               |    Graph ${GraphClass.Persons.id} {
+               |      ?personId a schema:Person;
+               |                schema:sameAs ?personSameAs.
+               |      ?personSameAs schema:additionalType ${Person.gitLabSameAsAdditionalType};
+               |                    schema:identifier ${criteria.authUser.id.value}.
+               |    }
+               |
+               |    {
+               |      SELECT DISTINCT
+               |        ${v.datasetSameAs}
+               |        (GROUP_CONCAT(DISTINCT ?creatorName; separator=',') AS ${v.creatorNames})
+               |        (GROUP_CONCAT(DISTINCT ?keyword; separator=',') AS ${v.keywords})
+               |        (GROUP_CONCAT(DISTINCT ?encodedImageUrl; separator=',') AS ${v.images})
+               |      WHERE {
+               |        Graph schema:Dataset {
```

Comment:
Should we maybe use the `GraphClass.Datasets.id` as graph id here (and in few other places in the query)?

Reply:
Yes ofc! Forgot these occurrences to replace. Thanks